.github/workflows/unit_tests.yml

```
@@ -43,10 +43,19 @@ jobs:
         shell: bash -el {0}
         run: |
           flake8 --count hnn_core
+      - name: Test MPI
+        shell: bash -el {0}
+        run: |
+          if [ "${{ matrix.os }}" == "macos-latest" ]; then
+            export DYLD_FALLBACK_LIBRARY_PATH=${CONDA_PREFIX}/lib
+          fi;
+          mpiexec -np 2 nrniv -mpi -python -c 'from neuron import h; from mpi4py import MPI; \
+                                               print(f"Hello from proc {MPI.COMM_WORLD.Get_rank()}"); \
+                                               h.quit()'
```

What happens if you instruct openmpi to use threads instead of cores as its "processing elements"?

```suggestion
          mpiexec -np 2 -use-hwthread-cpus nrniv -mpi -python -c 'from neuron import h; from mpi4py import MPI; \
                                               print(f"Hello from proc {MPI.COMM_WORLD.Get_rank()}"); \
                                               h.quit()'
```

---

I'll try it out!

---

It didn't make a difference--same MPI Test result where it seems to be duplicating as single process.

---

Okay, thanks for giving it a try!